lua/neo-tree/sources/filesystem/lib/fs_scan.lua

```diff
@@ -106,6 +106,26 @@ local render_context = function(context)
   context = nil
 end
 
+local job_complete_async = function(context)
+  local state = context.state
+  local parent_id = context.parent_id
+  if #context.all_items == 0 then
+    log.info("No items, skipping git ignored/status lookups")
+  elseif state.filtered_items.hide_gitignored or state.enable_git_status then
+    local mark_ignored_async = async.wrap(function (_state, _all_items, _callback)
+      git.mark_ignored(_state, _all_items, _callback)
+    end, 3)
+    local all_items = mark_ignored_async(state, context.all_items)
```

Comment:
**info** here I am always using async version of git_ignore, in contrary to `job_complete` that respects `require("neo-tree").config.git_status_async` and acts accordingly. 

I could merge these two together, but I am hesitant as the old one calls `render_context` which in the context of my function is unnecessary (at least in this place). 

Nevertheless, I still can add `config.git_status_asyn` to `job_complete_async` though I am not sure what are the benefits of that as everything is "async" in this callstack. Wdyt?

Reply:
I don't remember how everything works off the top of my head and I don't think I have enough time to dig in now, but I can say that if the filesystem is being loaded async it probably doesn't make sense to load git_status synchronously. I can see the other way around where the fs is scanned sync but git status is async.

Reply:
That sounds reasonable :+1: